### PR TITLE
Update frequency parsing in Gaussian

### PR DESCRIPTION
**Pull request by ndaelman-hu**
Limit extraction of vibrational analysis to '--' header. Also update schema and add dynamic unit extraction.

### CODE REVIEW

File: electronicparsers/gaussian/parser.py [hunk 1]
```diff
@@ -100,6 +100,25 @@ def str_to_force_constants(val_in):
             fc = fc + fc.T - np.diag(fc.diagonal())
             return fc
 
+        def str_to_units(unit: str):
```
Comment (ladinesa) [inline electronicparsers/gaussian/parser.py]: a map would be more desirable here instead of ifs

**Review by ladinesa — APPROVED** [review]

File: electronicparsers/gaussian/parser.py (hunk 1 above)
Comment (ndaelman-hu) [reply electronicparsers/gaussian/parser.py]: I agree, is a bit nicer to look at. Will update.

**Comment by ndaelman-hu** [comment]
@JosePizarro3 my test is failing at Yambo GW and TBStudio... Did something break there, cause I only touched Gaussian?

**Comment by ndaelman-hu** [comment]
Note: won't add testing. Have made a memo in #233 .

**Comment by ndaelman-hu** [comment]
> @JosePizarro3 my test is failing at Yambo GW and TBStudio... Did something break there, cause I only touched Gaussian?

So, these errors come from the updated type checking and are NOT caused by this PR.
@JosePizarro3 and @ladinesa I leave it up to you to fix those at your discretion. This PR will be merged.

**Pull request merged**